packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.types.ts

```diff
@@ -81,6 +81,18 @@ export interface IMessageBarProps extends React.HTMLAttributes<HTMLElement> {
    * Call to provide customized styling that will layer on top of the variant rules.
    */
   styles?: IStyleFunctionOrObject<IMessageBarStyleProps, IMessageBarStyles>;
+
+  /**
+   * Custom icon to replace the dismiss icon.
```

Comment:
Interesting. I'm not so enthusiastic about allowing this because it implies that we should do the same on all the other components with close buttons (not going to ask you to do that, but it opens the possibility), and I'm concerned that it sort of invites bad UX patterns (like using something besides an x for a close button). So I'll have to talk to some people about this tomorrow and get back to you.

Reply:
Sounds good! My only argument against what we do today is that the component has a dependency on a specific close icon that no one would know about unless they look through the code or happen to map all of Fabric's icon. 

I don't actually know much about icon mapping process, but I assume its some extra steps that needs to be taken by the consumer.